### PR TITLE
Get rid of json_encode_dict; fixes #89

### LINKED ISSUE #89: Empty config
Hello

on debian strech and saltstack 2018.3.2+ds-1 generate empty config

```
# OpenVPN server configuration 'xxxx'
# Managed by Salt
# Template: salt://openvpn/files/server.jinja


; Networking
proto udp


; Operating system
dev tun
user nobody
group nogroup
persist-key
persist-tun


; Data transfer


; Logging / Debugging
verb 3


; Cryptographic parameters
# ca is not set !
# cert is not set !
# key is not set !

; check with https://bettercrypto.org/static/applied-crypto-hardening.pdf

; Certificate verification
; (Anti MITM)


; Up/Down Scripts


tls-server
port 1194
```

i must remove in file `openvpn/config.sls` on line 37 ([this](https://github.com/saltstack-formulas/openvpn-formula/blob/master/openvpn/config.sls#L37)) json_encode_dict  and now config for server is generate OK.

```yaml
config:
`openvpn:
  lookup:
    dh_files: ['4096']
  server:
    name:
      port: 1194
      proto: udp
      topology: subnet
      ca: /etc/openvpn/ca.crt
      ca_content: |
        -----BEGIN CERTIFICATE-----
        -----END CERTIFICATE-----
      cert: /etc/openvpn/cert.pem
      cert_content: |
        -----BEGIN CERTIFICATE-----
        -----END CERTIFICATE-----
      key: /etc/openvpn/key.pem
      key_content: |
        -----BEGIN PRIVATE KEY-----
        -----END PRIVATE KEY-----
      dh: dh4096.pem
      server: 'x.0.0.0 255.255.255.0'
      ifconfig_pool_persist: ipp.txt
      push:
        - "route x.0.0.0 255.255.255.0"
      client_config:
        xxxxx: |
          ifconfig-push x.0.0.x 255.255.255.0
      client_to_client: False
      tls_auth: /etc/openvpn/tls.key 0
      tls_crypt: /etc/openvpn/tls.key
      tls_version_min: 1.2
      ta_content: |
        -----BEGIN OpenVPN Static key V1-----
        -----END OpenVPN Static key V1-----
      comp_lzo:
      user: nobody
      group: nogroup
      persist_key:
      persist_tun:
      status: /var/log/openvpn/openvpn-status.log
      log: /var/log/openvpn/openvpn.log
      client_config_dir: clients
`
```

### PR DESCRIPTION
**Pull request by alxwr**
Tested on

- Ubuntu 18.04
- FreeBSD 11.2

This PR should fix #89.

### CODE REVIEW

**Comment by javierbertoli** [comment]
@alxwr my only 'concern' is that the previous passing of `config` from the state file to the template performed [some 'sanitizing' on the valid sections accepted](https://github.com/saltstack-formulas/openvpn-formula/blob/f2440c6b2b3277ae767d53f41a20c281941e087f/openvpn/config.sls#L8-L9) and only the config parameters for the desired setup (`['client', 'server', 'peer']`) is passed, while in this PR all are passed at once, with unexpected results, am I right? Or am I reading it wrong?

**Comment by alxwr** [comment]
@javierbertoli When the template is rendered in [Line 30](https://github.com/saltstack-formulas/openvpn-formula/blob/f2440c6b2b3277ae767d53f41a20c281941e087f/openvpn/config.sls#L30) it is constrained by the context defined in [Lines 34 to 38](https://github.com/saltstack-formulas/openvpn-formula/blob/f2440c6b2b3277ae767d53f41a20c281941e087f/openvpn/config.sls#L34) which derives its values (partly) from [Lines 7 to 9](https://github.com/saltstack-formulas/openvpn-formula/blob/f2440c6b2b3277ae767d53f41a20c281941e087f/openvpn/config.sls#L7).

Long story short: the template calls Pillar directly but with the same keys that are used when `config` is set by the second `for` loop.

**Comment by aboe76** [comment]
@javierbertoli are you happy with the explanation?

**Comment by javierbertoli** [comment]
@aboe76 yeah, got it now. LGTM, I think we're ok to merge.

**Comment by aboe76** [comment]
@alxwr and @javierbertoli thanks for the work.

**Comment by alxwr** [comment]
@javierbertoli and @aboe76 thans for the review!
